### PR TITLE
Trace upd

### PR DESCRIPTION
**Pull request by snhenson**
##### Description of change

This extends the trace output to support some additional features and TLS 1.3 NewSessionTicket format.

Note: trace should be moved to a PACKET at some point, it's a bit messy.

### CODE REVIEW

File: ssl/t1_trce.c [hunk 1]
```diff
@@ -683,6 +695,24 @@ static int ssl_print_extension(BIO *bio, int indent, int server,
         if (extlen != xlen + 2)
             return 0;
         return ssl_trace_list(bio, indent + 2, ext + 2, xlen, 2, ssl_groups_tbl);
+    case TLSEXT_TYPE_application_layer_protocol_negotiation:
+        if (extlen < 2)
+            return 0;
+        xlen = (ext[0] << 8) | ext[1];
+        if (extlen != xlen + 2)
+            return 0;
+        ext += 2;
```
Comment (richsalz) [inline ssl/t1_trce.c]: Maybe turn this into a for loop and put line 712 into the statement as well?  Yes doing that would mean plen decl moves up.  Just a suggestion.

File: ssl/t1_trce.c [hunk 2]
```diff
@@ -428,6 +428,10 @@ static ssl_trace_tbl ssl_ciphers_tbl[] = {
     {0xCCAD, "TLS_DHE_PSK_WITH_CHACHA20_POLY1305"},
     {0xCCAE, "TLS_RSA_PSK_WITH_CHACHA20_POLY1305"},
     {0x1301, "TLS_AES_128_GCM_SHA256"},
+    {0x1302, "TLS_AES_256_GCM_SHA384"},
```
Comment (richsalz) [inline ssl/t1_trce.c]: Okay, not for this PR, but I'm gonna gripe.  We have all these ```_TXT_``` names and we're not using them.  And why aren't we using the cipher constants?  Again, not for this PR.

File: ssl/t1_trce.c [hunk 3]
```diff
@@ -683,6 +695,24 @@ static int ssl_print_extension(BIO *bio, int indent, int server,
...
         return ssl_trace_list(bio, indent + 2, ext + 2, xlen, 2, ssl_groups_tbl);
+    case TLSEXT_TYPE_application_layer_protocol_negotiation:
+        if (extlen < 2)
+            return 0;
+        xlen = (ext[0] << 8) | ext[1];
+        if (extlen != xlen + 2)
+            return 0;
+        ext += 2;
+        while (xlen > 0) {
+            size_t plen = *ext++;
```
Comment (richsalz) [inline ssl/t1_trce.c]: blank line after decl.

File: ssl/t1_trce.c [hunk 4]
```diff
@@ -1251,8 +1281,23 @@ static int ssl_print_ticket(BIO *bio, int indent,
     msg += 4;
     BIO_indent(bio, indent + 2, 80);
     BIO_printf(bio, "ticket_lifetime_hint=%u\n", tick_life);
+    if (SSL_IS_TLS13(s)) {
+        unsigned int ticket_age_add;
```
Comment (richsalz) [inline ssl/t1_trce.c]: blank line.  sigh, maybe time to change the coding style for decls not at start of function?

Comment (snhenson) [inline ssl/t1_trce.c]: Sorry it been doing things without a blank line for years and I backslide into my old ways occasionally...

File: ssl/t1_trce.c (hunk 2 above)
Comment (snhenson) [inline ssl/t1_trce.c]: Well there are a couple of reasons. This table has the official (as defined in spec) names for the ciphersuites which don't always agree with our historical versions, where the two agree we could use the _TXT_ version. The cipher constants, yeah we could use those where they exist (this table also includes ciphersuites we don't support though for debugging purposes.

BTW that table was originally generated from a script which extracted everthing from the relevant specs.

**Comment by snhenson** [comment]
Update: there were missing blank lines all over the place in t1_trce.c, added them. Also added a couple of minor cosmetic improvements.

**Pull request closed without merging**